File: Software/SoftwareIdeasModeler/Data/Templates/Code/CSharp/FluentNHibernate/Class.cs
<%==SIM:ExplicitLinesOn==%>
<%==SIM:Set:private=private==%>
<%==SIM:Set:protected=protected==%>
<%==SIM:Set:package=internal==%>
<%==SIM:Set:public=public==%>
using NHibernate;<%==SIM:Line==%>
using NHibernate.Criterion;<%==SIM:Line(2)==%>

namespace <%==SIM:Element.Namespace==%><%==SIM:Line==%>
{<%==SIM:Line==%>
<%==SIM:ForEach:Element.DocumentationLines==%>//<%==SIM:DocumentationLine==%><%==SIM:Line==%>
<%==SIM:EndFor==%>
    public class <%==SIM:Element.Name==%><%==SIM:Line==%>
    {<%==SIM:Line==%>
<%==SIM:ForEach:Element.AllAttributes==%>
        public virtual <%==SIM:Attribute.TypeName==%><%==SIM:IfNot:Attribute.ReplacedType.IsNullable==%><%==SIM:If:Attribute.IsNullable==%>?<%==SIM:EndIf==%><%==SIM:EndIf==%> <%==SIM:Attribute.Name==%> { get; set; }
<%==SIM:Line==%>
<%==SIM:EndFor==%>

<%==SIM:ForEach:Element.InverseReferences==%>
        public virtual IList<<%==SIM:Reference.ForeignKeyEntity.Name==%>> <%==SIM:Reference.Name==%>;
<%==SIM:Line==%>
<%==SIM:EndFor==%>
<%==SIM:Line==%>
        public <%==SIM:Element.Name==%>()<%==SIM:Line==%>
        {<%==SIM:Line==%>
<%==SIM:ForEach:Element.InverseReferences==%>
            <%==SIM:Reference.Name==%> = new List<<%==SIM:Reference.ForeignKeyEntity.Name==%>>();
<%==SIM:Line==%>
<%==SIM:EndFor==%>
        }<%==SIM:Line==%>
<%==SIM:Line==%>

<%==SIM:If:GreaterThan(Element.PrimaryKeys.Count,1)==%>
        public override bool Equals(object obj)<%==SIM:Line==%>
        {<%==SIM:Line==%>
            if (obj == null || GetType() != obj.GetType()) return false;<%==SIM:Line==%>
            <%==SIM:Element.Name==%> other = (<%==SIM:Element.Name==%>)obj;<%==SIM:Line==%><%==SIM:Line==%>
            if (<%==SIM:ForEach:Element.PrimaryKeys==%> this.<%==SIM:Key.Name==%> == other.<%==SIM:Key.Name==%><%==SIM:IfNot:IsLastItem==%> && <%==SIM:EndIf==%><%==SIM:EndFor==%>)<%==SIM:Line==%>
                return true;<%==SIM:Line==%>
            else<%==SIM:Line==%>
                return false;<%==SIM:Line==%>
        }<%==SIM:Line==%>
        <%==SIM:Line==%>
        public override int GetHashCode()<%==SIM:Line==%>
        {<%==SIM:Line==%>
            return <%==SIM:ForEach:Element.PrimaryKeys==%> this.<%==SIM:Key.Name==%>.GetHashCode()<%==SIM:IfNot:IsLastItem==%> + <%==SIM:EndIf==%><%==SIM:EndFor==%>;
            <%==SIM:Line==%>
        }<%==SIM:Line==%>
<%==SIM:EndIf==%>
<%==SIM:Line==%>

    }<%==SIM:Line==%>
}<%==SIM:Line==%>

File: Software/SoftwareIdeasModeler/Data/Templates/Code/VbNet/FluentNHibernate/Class.vb
<%==SIM:ExplicitLinesOn==%>
<%==SIM:Set:private=Private==%>
<%==SIM:Set:protected=Protected==%>
<%==SIM:Set:package=Friend==%>
<%==SIM:Set:public=Public==%>
Imports NHibernate<%==SIM:Line==%>
Imports NHibernate.Criterion<%==SIM:Line(2)==%>

Namespace <%==SIM:Element.Namespace==%><%==SIM:Line(2)==%>
<%==SIM:ForEach:Element.DocumentationLines==%>'<%==SIM:DocumentationLine==%><%==SIM:Line==%>
<%==SIM:EndFor==%>
    Public Class <%==SIM:Element.Name==%><%==SIM:Line(2)==%>

<%==SIM:ForEach:Element.AllAttributes==%>
        Public Overridable Property <%==SIM:Attribute.Name==%> As <%==SIM:Attribute.TypeName==%><%==SIM:IfNot:Attribute.ReplacedType.IsNullable==%><%==SIM:If:Attribute.IsNullable==%>?<%==SIM:EndIf==%><%==SIM:EndIf==%>
<%==SIM:Line==%>
<%==SIM:EndFor==%>

<%==SIM:ForEach:Element.InverseReferences==%>
        Public Overridable Property <%==SIM:Reference.Name==%> As IList(Of <%==SIM:Reference.ForeignKeyEntity.Name==%>) = New List(Of <%==SIM:Reference.ForeignKeyEntity.Name==%>)             
<%==SIM:Line==%>
<%==SIM:EndFor==%>
<%==SIM:Line==%>

<%==SIM:If:GreaterThan(Element.PrimaryKeys.Count,1)==%>
        Public Overrides Function Equals(ByVal obj As Object) As Boolean<%==SIM:Line==%>
            Dim other As <%==SIM:Element.Name==%> = CType(obj, <%==SIM:Element.Name==%>)<%==SIM:Line==%>
            If <%==SIM:ForEach:Element.PrimaryKeys==%> Me.<%==SIM:Key.Name==%> = other.<%==SIM:Key.Name==%><%==SIM:IfNot:IsLastItem==%> AndAlso <%==SIM:EndIf==%><%==SIM:EndFor==%> Then<%==SIM:Line==%>
                Return True<%==SIM:Line==%>
            Else<%==SIM:Line==%>
                Return False<%==SIM:Line==%>
            End If<%==SIM:Line==%>
        End Function<%==SIM:Line==%>
        <%==SIM:Line==%>
        Public Overrides Function GetHashCode() As Integer<%==SIM:Line==%>
            Dim hash As Long<%==SIM:Line==%>

            hash = <%==SIM:ForEach:Element.PrimaryKeys==%> CLng(Me.<%==SIM:Key.Name==%>.GetHashCode())<%==SIM:IfNot:IsLastItem==%> + <%==SIM:EndIf==%><%==SIM:EndFor==%>
            <%==SIM:Line==%><%==SIM:Line==%>
            Return hash Mod Integer.MaxValue<%==SIM:Line==%>
        End Function<%==SIM:Line==%>
<%==SIM:EndIf==%>
<%==SIM:Line==%>

    End Class<%==SIM:Line==%>
End Namespace<%==SIM:Line==%>

